plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php

```diff
@@ -150,3 +150,25 @@
 		<?php endif; ?>
 	</div>
 </div>
+<script type="text/template" id="tmpl-wc-modal-set-price-variations">
+	<div class="wc-backbone-modal">
+		<div class="wc-backbone-modal-content">
+			<div class="components-modal__content woocommerce-set-price-variations" role="document">
+				<div class="components-modal__header">
+					<h2><?php echo esc_attr( $modal_title ); ?></h2>
+				</div>
+				<div class="woocommerce-usage-modal__wrapper">
+					<div class="woocommerce-usage-modal__message">
+						<span>Add price to all variations that don't have a price (<?php echo esc_attr( get_woocommerce_currency_symbol() ); ?> <?php echo esc_textarea( get_woocommerce_currency() ); ?>)</span>
+						<input type="text" class="components-text-control__input wc_input_variations_price"/>
+					</div>
+					<div class="woocommerce-usage-modal__actions">
+						<button class="modal-close components-button is-secondary"><?php esc_html_e( 'Cancel', 'woocommerce' ); ?></button>
+						<button class="modal-close button components-button add_variations_price_button is-primary" disabled><?php esc_html_e( 'Add prices', 'woocommerce' ); ?></button>
```

Comment:
Does PHP provide a helper function for `esc_html_e` for multiple as well?
It might be nice to show either `Add price` or `Add prices` depending on how many are missing their price.
Same in the notice, the notice might be easier to do given you already have access to the count.

This is not a must, but might be nice, if it's not to difficult.

Reply:
Hi @louwie17,

This change is not trivial and it would imply calling the database again.

The number of variations without price is calculated [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/wc-admin-functions.php#L479-L497). Because [html-product-data-variations.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-variations.php) is being loaded before [wc-admin-functions.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/wc-admin-functions.php) we're not able to use a `global` variable, since the value is calculated after the first file is processed.

I'm not sure if the change is worth the effort. What do you think about merging this PR as it is and maybe tackling that change in a follow-up PR?

Reply:
A follow-up PR sounds good with me.